tests/backup/backup_azure_credential_test.go

```diff
@@ -0,0 +1,216 @@
+		sourceClusterUid                                       string
+		backupNameWithBkpLocationHavingMandatoryParameters     string
+		restoreNameWithBkpLocationHavingMandatoryParameters    string
+		backupNameWithBkpLocationHavingAllParameters           string
+		restoreNameWithBkpLocationHavingAllParameters          string
+		backupNameWithBkpLocationHavingOnlyMandatoryParameters string
+		restoreNameWithBkpLocationHavingOnlyMandatoryFields    string
+		credUidWithMandatoryFields                             string
+		azureCredNameWithAllFields                             string
+		azureCredNameWithMandatoryFields                       string
+		azureBackupLocationNameWithAllFields                   string
+		azureBackupLocationNameWithMandatoryFields             string
+		backupLocationAllFieldsUID                             string
+		backupLocationMandatoryFieldsUID                       string
+		appNamespaces                                          []string
+		controlChannel                                         chan string
+		errorGroup                                             *errgroup.Group
+		azureConfigFields                                      *api.AzureConfig
+		scheduledAppContexts                                   []*scheduler.Context
+		contexts                                               []*scheduler.Context
+		appContexts                                            []*scheduler.Context
+	)
+
+	backupLocationMap1 := make(map[string]string)
+	cloudCredentialMap1 := make(map[string]string)
+	backupLocationMap2 := make(map[string]string)
+	cloudCredentialMap2 := make(map[string]string)
+	JustBeforeEach(func() {
+		StartPxBackupTorpedoTest("AzureCloudAccountCreationWithMandatoryAndNonMandatoryFields", "Azure cloud account with mandatory and non mandatory fields", nil, 31661, Sagrawal, Q2FY25)
+		log.InfoD("Deploying applications required for the testcase")
+		contexts = make([]*scheduler.Context, 0)
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			taskName := fmt.Sprintf("%s-%d", TaskNamePrefix, i)
+			appContexts = ScheduleApplications(taskName)
+			contexts = append(contexts, appContexts...)
+			for _, ctx := range appContexts {
+				ctx.ReadinessTimeout = AppReadinessTimeout
+				namespace := GetAppNamespace(ctx, taskName)
+				appNamespaces = append(appNamespaces, namespace)
+				scheduledAppContexts = append(scheduledAppContexts, ctx)
+			}
+		}
+	})
+
+	It("Azure cloud account with mandatory and non mandatory fields", func() {
+		ctx, err := backup.GetAdminCtxFromSecret()
+		log.FailOnError(err, "Fetching px-central-admin ctx")
+		Step("Validating applications", func() {
+			log.InfoD("Validating applications")
+			ctx, _ := backup.GetAdminCtxFromSecret()
+			controlChannel, errorGroup = ValidateApplicationsStartData(scheduledAppContexts, ctx)
+		})
+
+		Step("Creating azure cloud account and backup location with mandatory and non-mandatory fields", func() {
```

Comment:
I still think it is better to split this step. First step to create azure cloud account and backup location with mandatory fields
second step with optional fields

Reply:
done